front/src/applications/operationalStudies/hooks/useSimulationResults.ts

```diff
@@ -18,50 +27,138 @@ const useSimulationResults = (): SimulationResultsData => {
+  );
+
+  const { data: selectedPacedTrain } = osrdEditoastApi.endpoints.getPacedTrainById.useQuery(
+    {
+      id: editoastSelectedTrainId!,
+    },
+    { skip: !editoastSelectedTrainId || (selectedTrainId && !isOccurrence(selectedTrainId)) }
   );
 
-  const { data: rawPath } = osrdEditoastApi.endpoints.getTrainScheduleByIdPath.useQuery(
+  const { data: rawTrainSchedulePath } =
+    osrdEditoastApi.endpoints.getTrainScheduleByIdPath.useQuery(
+      {
+        id: editoastSelectedTrainId!,
+        infraId: infraId!,
+      },
+      {
+        skip:
+          !editoastSelectedTrainId ||
+          !infraId ||
+          (selectedTrainId && !isTrainSchedule(selectedTrainId)),
+      }
+    );
+
+  const { data: rawPacedTrainPath } = osrdEditoastApi.endpoints.getPacedTrainByIdPath.useQuery(
     {
       id: editoastSelectedTrainId!,
       infraId: infraId!,
     },
-    { skip: !editoastSelectedTrainId || !infraId }
+    {
+      skip:
+        !editoastSelectedTrainId || !infraId || (selectedTrainId && !isOccurrence(selectedTrainId)),
+    }
   );
-  const path = selectedTrainId && rawPath?.status === 'success' ? rawPath : undefined;
+  const path = useMemo(() => {
+    if (!selectedTrainId) return undefined;
 
-  const { data: trainSimulation } =
+    if (isTrainSchedule(selectedTrainId)) {
+      return rawTrainSchedulePath?.status === 'success' ? rawTrainSchedulePath : undefined;
+    }
+    return rawPacedTrainPath?.status === 'success' ? rawPacedTrainPath : undefined;
+  }, [selectedTrainId, rawTrainSchedulePath, rawPacedTrainPath]);
+
+  const { data: selectedTrainScheduleSimulation } =
     osrdEditoastApi.endpoints.getTrainScheduleByIdSimulation.useQuery(
       { id: editoastSelectedTrainId!, infraId: infraId!, electricalProfileSetId },
-      { skip: !editoastSelectedTrainId || !infraId }
+      {
+        skip:
+          !editoastSelectedTrainId ||
+          !infraId ||
+          (selectedTrainId && !isTrainSchedule(selectedTrainId)),
+      }
+    );
+
+  const { data: selectedPacedTrainSimulation } =
+    osrdEditoastApi.endpoints.getPacedTrainByIdSimulation.useQuery(
+      { id: editoastSelectedTrainId!, infraId: infraId!, electricalProfileSetId },
+      {
+        skip:
+          !editoastSelectedTrainId ||
+          !infraId ||
+          (selectedTrainId && !isOccurrence(selectedTrainId)),
+      }
     );
 
+  const selectedTimetableItemSimulationData = useMemo(() => {
+    if (!selectedTrainId) return undefined;
+    if (isTrainSchedule(selectedTrainId) && !selectedTrainSchedule) return undefined;
+    if (isOccurrence(selectedTrainId) && !selectedPacedTrain) return undefined;
+
+    if (isTrainSchedule(selectedTrainId)) {
+      return {
+        selectedTimetableItem: selectedTrainSchedule,
+        selectedTimetableItemSimulation: selectedTrainScheduleSimulation,
+        selectedTimetableItemStartTime: selectedTrainSchedule?.start_time,
+      };
+    }
```

Comment:
```suggestion
    if (isTrainSchedule(selectedTrainId)) {
      return selectedTrainSchedule
        ? {
            selectedTimetableItem: selectedTrainSchedule,
            selectedTimetableItemSimulation: selectedTrainScheduleSimulation,
            selectedTimetableItemStartTime: selectedTrainSchedule?.start_time,
          }
        : undefined;
    }

    if (!selectedPacedTrain) return undefined;
```
Then can remove some `!` and typing (`| undefined`)

Reply:
Oh yes nice ! Done